### PR TITLE
[#74] integration tests

### LINKED ISSUE #74: Replace tests with local server and Unit-Tests
Downloads should not be done from random internet locations. Instead, start jetty (or any other local server) and download from there.

### PR DESCRIPTION
**Pull request by bmarwell**
Speed up builds considerably by using a local server.

fixes #74 

Still in progress

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -183,6 +183,11 @@
       <artifactId>slf4j-api</artifactId>
       <version>1.7.9</version>
     </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-simple</artifactId>
+      <version>1.7.9</version>
```
Comment (slawekjaranowski) [inline pom.xml:189]: scope test

File: src/main/java/org/mojohaus/mojo/wagon/AbstractCopyMojo.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package org.codehaus.mojo.wagon;
+package org.mojohaus.mojo.wagon;
```
Comment (slawekjaranowski) [inline src/main/java/org/mojohaus/mojo/wagon/AbstractCopyMojo.java:1]: maybe package change in separate PR

File: pom.xml (hunk 1 above)
Comment (bmarwell) [reply pom.xml:189]: Thanks!

File: src/main/java/org/mojohaus/mojo/wagon/AbstractCopyMojo.java (hunk 2 above)
Comment (bmarwell) [reply src/main/java/org/mojohaus/mojo/wagon/AbstractCopyMojo.java]: Local test which I wouldn't have wanted to commit. Thanks for catching!

File: pom.xml [hunk 3]
```diff
@@ -297,6 +322,21 @@
...
         </executions>
       </plugin>
+
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-failsafe-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>integration-test</id>
+            <phase>integration-test</phase>
```
Comment (slawekjaranowski) [inline pom.xml:332]: Default phases for goals are ok, we don't need override it

File: .github/workflows/maven.yml [hunk 4]
```diff
@@ -24,8 +24,8 @@ jobs:
     name: Verify
     uses: apache/maven-gh-actions-shared/.github/workflows/maven-verify.yml@v3
     with:
-      ff-goal: '-P run-its install'         # site use project version for reporting
-      verify-goal: '-P run-its install'     # should be the same as for first build
+      ff-goal: '-ntp -P run-its install'         # site use project version for reporting
+      verify-goal: '-ntp -P run-its install'     # should be the same as for first build
```
Comment (slawekjaranowski) [inline .github/workflows/maven.yml:28]: verify should be enough for this project, also there is not profile run-it here

**Review by slawekjaranowski — APPROVED** [review]

**Pull request merged**